src/Aimeos/Shop/Controller/CatalogController.php

```diff
@@ -51,8 +51,13 @@ public function detailAction()
 	{
 		foreach( app( 'config' )->get( 'shop.page.catalog-detail' ) as $name )
 		{
-			$params['aiheader'][$name] = Shop::get( $name )->getHeader();
-			$params['aibody'][$name] = Shop::get( $name )->getBody();
+			try {
+                $params['aiheader'][$name] = Shop::get( $name )->getHeader();
+			    $params['aibody'][$name] = Shop::get( $name )->getBody();
+            } catch(\Exception $e) {
+                if ( $e->getCode() === 404 ) { abort(404); }
+                throw $e;
+            }
```

Comment:
Can you please fix indentions (one tab per indention)?

Reply:
Fixed